docker/marin/Dockerfile.cluster

```diff
@@ -67,11 +67,18 @@ RUN conda install -c conda-forge ncurses -y
 
 # Install Jax for TPU usage, fsspec for GCS access, and transformers for OLMO
 WORKDIR /tmp/
-RUN pip install --no-cache-dir -U pip setuptools wheel
-RUN pip install --no-cache-dir torch==2.7.0 jax[tpu]==0.5.1 -f https://download.pytorch.org/whl/nightly/cpu -f https://storage.googleapis.com/libtpu-releases/index.html
+RUN pip install --no-cache-dir -U pip setuptools wheel uv
+
+# Create an isolated virtual environment with uv and use its uv for installations
+RUN uv venv /opt/marin/venv
+ENV VIRTUAL_ENV=/opt/marin/venv
+ENV PATH=/opt/marin/venv/bin:$PATH
+
+# Install core ML libraries inside the created virtual environment
+RUN uv pip install --no-cache-dir torch==2.7.0 jax[tpu]==0.5.1 -f https://download.pytorch.org/whl/nightly/cpu -f https://storage.googleapis.com/libtpu-releases/index.html
 COPY pyproject.toml /tmp/
 # re-specify jax/torch/torch_xla to ensure they still work if we bump versions in pyproject.toml
-RUN pip install --no-cache-dir -e . -f https://download.pytorch.org/whl/cpu jax[tpu] torch -f https://storage.googleapis.com/jax-releases/libtpu_releases.html
+RUN uv pip install --no-cache-dir -e . -f https://download.pytorch.org/whl/cpu jax[tpu] torch -f https://storage.googleapis.com/jax-releases/libtpu_releases.html
```

Comment:
Are we switching over to uv for everything?

Reply:
there's a neat docker setup for UV that does caching and partial updates pretty well fyi

https://docs.astral.sh/uv/guides/integration/docker/